src/components/connect.js

```diff
@@ -19,6 +19,14 @@ function getDisplayName(WrappedComponent) {
   return WrappedComponent.displayName || WrappedComponent.name || 'Component'
 }
 
+function getMapDispatch(value) {
+  switch (true) {
+    case (typeof value === 'function'): return value
```

Comment:
This is a very surprising use of `switch` statement. At this point I think the original version was clearer.